### PR TITLE
Workaround OOM issue in classification 2D integration tests

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
During debugging #3934 , we found that the `test_integration_classification_2d` integration test uses more than 14GB memory which may cause OOM in CI server sometimes.
Then I tried to do lots experiments to identify the root cause with this command:
`export CUDA_VISIBLE_DEVICES=$(python -m tests.utils); python -m tests.test_integration_classification_2d`
1. Run the latest MONAI code of dev branch in the **latest PyTorch docker**, 14GB memeory.
2. Run the latest MONAI code of dev branch in the **very old 21.02 PyTorch docker**, 4GB memeory.
3. **Installed latest PyTorch 1.11** in the very old 21.02 PyTorch docker, then run the latest MONAI dev code, 4GB memory.

So I think it seems the issue is not related to MONAI or PyTorch code, maybe some CUDA caching logic issue?

Then I tried to analyze the integration test and found that: when running the training, every process of DataLoader will occupy 839MB GPU memory and we set `num_workers=10` in the test, the total GPU memory is **14GB**:
image

But in the old 21.02 docker, only the main process occupies GPU memory, so it's only **4GB**:
image

I also tried to change other args of DataLoader, like: `pin_memory`, `persistent_workers`, etc. it doesn't change anything.

So this PR changed to `num_workers` from 10 to 1, it will much less memory and will fix the OOM issue in CI server.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
> Installed latest PyTorch 1.11 in the very old 21.02 PyTorch docker, then run the latest MONAI dev code, 4GB memory.

docker 21.02 is with cuda 11.2, which version of pytorch 1.11 did you install?

**Comment by Nic-Ma** [comment]
I am testing slightly older dockers, maybe we can temporarily roll back to an older docker to avoid changing test cases.

Thanks.

**Comment by Nic-Ma** [comment]
I tested below PyTorch dockers with MONAI dev code and the built-in PyTorch:
1. 22.02: 14GB
2. 22.01: 14GB
3. 21.12: 14GB
4. 21:10: 4GB, no dataloader process occupied GPU memory
5. 21:08: 4GB
6. 21:02: 4GB

I plan to temporarily roll back our MONAI docker to PyTorch 21.10, do you guys have any concerns?

Thanks in advance.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 jobs:
   integration-py3:
     container:
-      image: nvcr.io/nvidia/pytorch:21.12-py3  # CUDA 11.5
+      image: nvcr.io/nvidia/pytorch:21.10-py3  # CUDA 11.4
```
Comment (wyli) [inline .github/workflows/integration.yml:10]: this image wouldn't work with line 37 `torch==1.11.0+cu115` could you test?

Comment (Nic-Ma) [reply .github/workflows/integration.yml:10]: updated in the latest commit.

Thanks.

Comment (wyli) [reply .github/workflows/integration.yml:10]: this ci config will be used when this PR is merged to dev, so https://github.com/Project-MONAI/MONAI/runs/5570109568?check_suite_focus=true is still out-of-date, I think you can test locally of the commands here or have a temporary test like this one https://github.com/Project-MONAI/MONAI/blame/3b771bb8774b89e5ea630b9782070415a979202c/.github/workflows/integration.yml#L6-L8

Comment (Nic-Ma) [reply .github/workflows/integration.yml:10]: OK, I already tested below commands in the 21.10 docker locally:
```
root@apt-sh-ai:/workspace/data/medical/MONAI# pip install torch==1.11.0+cu113 torchvision==0.12.0+cu113 -f https://download.pytorch.org/whl/torch_stable.html
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Requirement already satisfied: torch==1.11.0+cu113 in /opt/conda/lib/python3.8/site-packages (1.11.0+cu113)
Requirement already satisfied: torchvision==0.12.0+cu113 in /opt/conda/lib/python3.8/site-packages (0.12.0+cu113)
Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.8/site-packages (from torch==1.11.0+cu113) (3.10.0.2)
Requirement already satisfied: requests in /opt/conda/lib/python3.8/site-packages (from torchvision==0.12.0+cu113) (2.26.0)
Requirement already satisfied: numpy in /opt/conda/lib/python3.8/site-packages (from torchvision==0.12.0+cu113) (1.21.2)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in /opt/conda/lib/python3.8/site-packages (from torchvision==0.12.0+cu113) (9.0.1)
Requirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/lib/python3.8/site-packages (from requests->torchvision==0.12.0+cu113) (2.0.0)
Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.8/site-packages (from requests->torchvision==0.12.0+cu113) (3.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.8/site-packages (from requests->torchvision==0.12.0+cu113) (1.26.7)
Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.8/site-packages (from requests->torchvision==0.12.0+cu113) (2021.5.30)
root@apt-sh-ai:/workspace/data/medical/MONAI# export CUDA_VISIBLE_DEVICES=$(python -m tests.utils); python -m tests.test_integration_classification_2d
----------
Epoch 1/4
epoch 1 average loss:0.7807
saved new best metric model
current epoch 1 current AUC: 0.9978 current accuracy: 0.9538 best AUC: 0.9978 at epoch 1
----------
Epoch 2/4
epoch 2 average loss:0.1623
...
```
And the test GPU memory is as expected:
image

Thanks.

**Review by wyli — APPROVED** [review]

**Comment by wyli** [comment]
confirming this works fine -- https://github.com/Project-MONAI/MONAI/runs/5576994468?check_suite_focus=true

**Pull request merged**